### PR TITLE
Add benchmarks for creating Strings from ASCII

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by Catfish-Man** [comment]
@swift-ci please smoke test

**Review by lorentey — APPROVED** [review]
This looks good to me. I had two comments, but given the existing benchmarks, it may be best to ignore them. (Local consistency may be more valuable than global in this case.)

File: benchmark/single-source/UTF8Decode.swift [hunk 1]
```diff
@@ -42,6 +42,18 @@ public let UTF8Decode = [
       name: "UTF8Decode_InitFromBytes_ascii",
       runFunction: run_UTF8Decode_InitFromBytes_ascii,
       tags: [.validation, .api, .String]),
+    BenchmarkInfo(
+      name: "UTF8Decode_InitFromData_ascii_as_ascii",
```
[the remark this answers is not in the document]
Comment (Catfish-Man) [reply benchmark/single-source/UTF8Decode.swift]: I felt keeping the prefix the same was worthwhile for being able to do something like "run all the benchmarks for X". The name does really suck though.

**Pull request merged**